tests/navigation.spec.ts

```diff
@@ -0,0 +1,89 @@
+import { test, Page } from '@playwright/test';
+import { MainText } from './page-objects/main-text';
+import { Navigation } from './page-objects/navigation';
+
+const setHeadingAndTestNavigation = async (page: Page, navigation: Navigation, heading: string): Promise<void> => {
+  const mainText = new MainText(page);
+  const text = 'To Ganymede and Titan';
+  await mainText.setText(text);
+  await mainText.selectText(text);
+  await mainText.setTextHeading(heading);
+  await navigation.clickLink(text);
```

Comment:
So I wonder here if this is actually clicking the element that you expect, as you are using the same text. I can see below that you call this 4 times in the fixture 'dynamic main text headings', but the text will be the same each time. Hence won't the 2nd, 3rd and 4th calls actually click the first element?
Appending the heading to the `text` variable might solve this, or add an additional parameter to customise the text for each call.

Reply:
each test runs in a separate browser instance so these wont interfere with each other